src/errors/test-run/templates.js

```diff
@@ -23,6 +23,27 @@ function markup (err, msgMarkup) {
     return msgMarkup;
 }
 
+function formatOrdinalNumber (number) {
+    if (number < 4 || number > 20) {
+        switch (number % 10) {
+        case 1: return number + 'st';
+        case 2: return number + 'nd';
+        case 3: return number + 'rd';
+        }
+    }
+
+    return number + 'th';
+}
+
+function formatArrayArgumentErrorReason (err) {
+    if (err.invalidLength)
+        return `its <code>length</code> property was <code>${err.actualValue}</code>`;
+    else if (err.elementIndex !== void 0)
+        return `its ${formatOrdinalNumber(err.elementIndex)} element was <code>${err.actualValue}</code>`;
+
+    return `it was <code>${err.actualValue}</code>`;
```

Comment:
This is not quite what I meant. I've asked why we needed two errors for non-empty array check:
https://github.com/DevExpress/testcafe/pull/508#discussion_r62314255 - `actionListArgumentError` expected non-empty array, and then we have `actionListArgumentLengthError` which is just check for non-empty error. Why we can't use `actionListArgumentError` in both cases? So here we need two errors: `actionListArgumentError` and error that checks that all elements in list has expected type.